### ISSUE TITLE
Element selection in large graphs

### ISSUE THREAD

**Issue by fedarko** [cytoscape/cytoscape.js#1723]
Hi,

In large graphs (upwards of around 1,000 nodes, 1,000 edges), I've noticed that selecting elements is somewhat slower than in smaller-sized graphs—which is to be expected. However, one of the stranger things I've noticed with large graphs like these is that box selection of a few elements is often faster than tap selection of a single element.

My guess (which could be wrong) is that this occurs due to the overlay applied upon elements in the `:active` state, which isn't used during box selection of elements. If this makes sense—would it be possible to add an optional feature in which tap selection just immediately marks a given element as selected, without any `:active` animation? Although the overlay does look nice, it seems to be making selection markedly slower for large graphs. I think this might make selecting individual elements in larger graphs a faster operation.

**Comment by maxkfranz** [comment]
You can override the `:active` state in your stylesheet

**Comment by fedarko** [comment]
Thanks for the response. Which styles for the `:active` state should I override to do this? As far as I can tell, I can change [the attributes](http://js.cytoscape.org/#style/overlay) of the overlay but not actually disable the overlay itself.

**Comment by maxkfranz** [comment]
Zero padding or zero opacity

**Comment by maxkfranz** [comment]
You could alternatively `cy.style().clear()` before you set your own stylesheet.  That will use no default stylesheet at all, so some properties (like for `core`) will be completely missing.  If you go that route, you're on your own w.r.t. missing style property errors --- but you'll have complete control over the entire stylesheet.

**Comment by fedarko** [comment]
Sorry to bother—I tried your first suggestion, and using zero padding/opacity for the `:active` state doesn't seem to result in a noticeable change in time taken to select an individual element. I'll look into using `cy.style().clear()` when I get some more time, but I'm thinking it might be easier to just modify Cytoscape.js to accomplish this.

For context (to hopefully be less vague), in Iceweasel (a slightly modified version of Firefox for Debian) version 38.7.1, the styling upon active nodes seems to be disabled upon tap selection regardless of the padding/opacity settings in place for `:active` elements. From a bit of looking around it seems like some of the version 2.7 performance enhancements mentioned in #1005 might be related to this. In Chromium (version 55.0.2883.75 on my machine), though, this styling is enabled by default. So when viewing the same region of the same graph in our application in both browsers, element selection in Iceweasel is much faster than in Chromium. Attaching two .gif files I recorded showing this (sorry for the image quality of the recording).

_Selection in Chromium: node is clicked at start of the gif and takes about 4-5 seconds to be selected. (The clock in the top-right of the screen can be used as a reference.)_
![chromium_individualselect1](https://cloud.githubusercontent.com/assets/4177727/23289706/a1176e1e-fa19-11e6-80bb-6ae7053e2366.gif)

_Selection in Iceweasel: node is clicked about 2 seconds after the start of the gif and takes about 2 seconds to be selected. (The clock in the top-right of the screen can be used as a reference.)_
![iceweasel_individualselect1](https://cloud.githubusercontent.com/assets/4177727/23289732/d9b8ca7e-fa19-11e6-88b8-34571699da55.gif)

Basically, I'm curious if this difference in styling for Iceweasel/Firefox could be applied to other browsers to speed up user interaction with the graph. If this isn't possible, or if the difference in styling isn't even intentional for some reason, that's fine—this isn't that significant of an issue for our application.

Thank you for your help!
